scipy/special/_basic.py

```diff
@@ -2036,13 +2036,15 @@ def factorial(n, exact=False):
     """
     if exact:
         if np.ndim(n) == 0:
-            return 0 if n < 0 else math.factorial(n)
+            return _factorial(n)
```

Comment:
Since we're in the exact branch you'll still want to use `math.factorial` here so that it handles very large integers.